agent/src/com/jonnyzzz/teamcity/virtual/run/docker/DockerVM.java

```diff
@@ -125,12 +139,14 @@ public BuildProcess startImpl() throws RunBuildException {
       }
 
       @NotNull
-      private List<String> scriptRun(@NotNull final File script) {
-        return Arrays.asList(
-                ctx.getShellLocation(),
-                "-c",
-                "\"source " + script.getName() + "\""
-        );
+      private List<String> scriptRun(@NotNull final File script) throws RunBuildException {
+        if(ctx.isDockerServerWindowsBased() && ctx.getShellLocationInsideContainer().equals("cmd.exe")) {
+          return  Arrays.asList(ctx.getShellLocationInsideContainer(), "/C", "call " + script.getName());
+        } else if(ctx.isDockerServerWindowsBased() && ctx.getShellLocationInsideContainer().equals("powershell.exe")) {
+          return  Arrays.asList(ctx.getShellLocationInsideContainer(), "-Command", "get-content " + script.getName() + " | Invoke-Expression");
```

Comment:
My first intention was to use `powershell.exe -File` but all files are creating with .bat extension. There is a TODO about it https://github.com/jonnyzzz/TeamCity.Virtual/blob/master/agent/src/com/jonnyzzz/teamcity/virtual/run/ScriptFile.java#L81
and I thought this workaround will fit. I will look to plugin to find out how to handle it properly.